sdk/monitor/azure-monitor-query/metrics-query-migration-guide.md

```diff
@@ -0,0 +1,315 @@
+
+This guide assists in migrating from the metrics functionality in `azure-monitor-query` to the dedicated `azure-monitor-query-metrics` library.
+
+## Table of contents
+
+- [Migration benefits](#migration-benefits)
+  - [Cross-service library improvements](#cross-service-library-improvements)
+- [Important changes](#important-changes)
+  - [Group ID, artifact ID, and package names](#group-id-artifact-id-and-package-names)
+  - [Client differences](#client-differences)
+  - [Instantiate metrics clients](#instantiate-metrics-clients)
+  - [Query metrics for multiple resources](#query-metrics-for-multiple-resources)
+  - [Setting audience for authentication](#setting-audience-for-authentication)
+- [Additional samples](#additional-samples)
+
+## Migration benefits
+
+The Azure Monitor Query library for Java has been modularized to provide more focused functionality. The metrics operations for querying multiple resources have been moved from the combined `azure-monitor-query` package to a dedicated `azure-monitor-query-metrics` package. This separation offers several advantages:
+
+- Smaller dependency footprint for applications that only need metrics functionality for multiple resources
+- More focused API design specific to metrics operations
+- Independent versioning allowing metrics functionality to evolve separately
+- Clearer separation of concerns between logs and metrics operations
+
+## Important changes
+
+### Group ID, artifact ID, and package names
+
+Both libraries have the same group ID (`com.azure`), but the artifact ID and package names have changed:
+
+- Previous artifact ID: `azure-monitor-query`
+- New artifact ID: `azure-monitor-query-metrics`
+
+- Previous package for metrics clients: `com.azure.monitor.query`
+- New package: `com.azure.monitor.query.metrics`
+
+**Maven dependency changes:**
+
+Previous dependency in `pom.xml`:
+```xml
+<dependency>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-monitor-query</artifactId>
+    <version>1.x.x</version>
+</dependency>
+```
+
+New dependency in `pom.xml`:
+```xml
+<dependency>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-monitor-query-metrics</artifactId>
+    <version>1.x.x</version>
+</dependency>
+```
+
+### Client differences
+
+It's important to note that the `azure-monitor-query` package contained two sets of metrics clients:
+
+1. `MetricsQueryClient`/`MetricsQueryAsyncClient`: For querying individual resource metrics, namespaces, or definitions from Azure Monitor
+2. `MetricsClient`/`MetricsAsyncClient`: For querying metrics from multiple resources at once
+
+The new `azure-monitor-query-metrics` library only includes the functionality previously provided by `MetricsClient`/`MetricsAsyncClient` for querying metrics from multiple resources. If you were using the `MetricsQueryClient`/`MetricsQueryAsyncClient` from the original package, you should migrate to usage the ARM library `azure-resourcemanager-monitor` package for that functionality.
```

Comment:
We should include samples to show how to migrate from `MetricsQueryClient` to the management library operations as well.

Reply:
The ARM samples are not straightforward and concise like the dataplane SDK samples, so I'll link out to them.